backend/tests/test_rate_limiting.py

```
@@ -0,0 +1,295 @@
+        
+        for method, endpoint, data in malformed_requests:
+            if method == 'POST':
+                response = client.post(
+                    endpoint,
+                    data=data,
+                    headers=auth_headers,
+                    content_type='application/json'
+                )
+            
+            # Should handle malformed requests gracefully
+            assert response.status_code in [400, 422], \
```

Pylint (3.3.7)

[error] Possibly using variable 'response' before assignment

(E0606)